File: todo-list/python/tests/task_builder.py
from datetime import date
from typing import Optional

from todo_list import Task


DEFAULT_ID = 1
DEFAULT_TITLE = ""


class TaskBuilder:
    def __init__(self) -> None:
        self._id = DEFAULT_ID
        self._title = DEFAULT_TITLE
        self._done = False
        self._due: Optional[date] = None

    def with_id(self, id: int) -> "TaskBuilder":
        self._id = id
        return self

    def titled(self, title: str) -> "TaskBuilder":
        self._title = title
        return self

    def due(self, due: date) -> "TaskBuilder":
        self._due = due
        return self

    def done(self) -> "TaskBuilder":
        self._done = True
        return self

    def build(self) -> Task:
        return Task(id=self._id, title=self._title, done=self._done, due=self._due)

File: todo-list/typescript/tests/taskBuilder.ts
import type { Task } from '../src/todoList.js';

const DEFAULT_ID = 1;
const DEFAULT_TITLE = '';

export class TaskBuilder {
  private _id = DEFAULT_ID;
  private _title = DEFAULT_TITLE;
  private _done = false;
  private _due: string | null = null;

  withId(id: number): this { this._id = id; return this; }
  titled(title: string): this { this._title = title; return this; }
  due(due: string): this { this._due = due; return this; }
  done(): this { this._done = true; return this; }

  build(): Task {
    return { id: this._id, title: this._title, done: this._done, due: this._due };
  }
}
